juju/juju
# Handle SIGTERM and stop hook on sidecar deploys.

Add the caasunitterminationworker to the caas sidecar unit agent manifold to handle receiving SIGTERM when running in
a container environment (k8s).

In the event a Pod is simply dying (expectation the unit will get a new Pod), then the worker will ask the uniter to shutdown, running the stop hook first before terminating the agent process.

In the event a Pod is dying because the Unit has reached the end of its life, then the termination worker just eats the SIGTERM allowing the Uniter to exit normally.

## QA steps

Deploy sidecar charm.
Install hook should run and start hook.
Delete pod.
Stop hook should run.
On restart the upgrade-charm hook should run then possibly config-changed hook then start hook.
Use juju to scale the application to 0.
Stop hook should run then the remove hook should run.

## Documentation changes

Task created to update/create documentation.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1926568
